### PR TITLE
[secrman] add arcade SECRMAN

### PR DESCRIPTION
**Pull request by israpps**
I will not refactor/reformat this code untill I have it fully reversed and functional

implementation is still incomplete.

### CODE REVIEW

File: iop/security/secrman/src/DongleAuth.c [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+            }
+            card_auth_60(port,slot);
+            mechacon_auth_80(cnum);
+          }
+        }
+      }
+    }
+  }
+  return 0;
+}
```
Comment (Florin9doi) [inline iop/security/secrman/src/DongleAuth.c:174]:
```suggestion
    if (GetMcCommandHandler() == NULL) {
        _arcade_printf("mcCommand isn't assigned\n");
        return 0;
    }
    memset(MechaChallenge2, 0, sizeof(MechaChallenge2));

    _arcade_printf("SecrAuthDongle start\n");
    if (!card_auth_60(port, slot))
        return 0;

    _arcade_printf("dongle auth 0x60\n");
    if (!mechacon_auth_80(cnum))
        return 0;

    _arcade_printf("mechacon auth 0x80\n");
    if (!card_auth(port, slot, 0xf0, 0)) {
        card_auth_60(port, slot);
        return 0;
    }

    _arcade_printf("dongle auth 0x00\n");
    if (!mechacon_auth_81(cnum | 0x40)) {
        card_auth_60(port, slot);
        return 0;
    }

    _arcade_printf("mechacon auth 0x81\n");
    if (!card_auth_read(port, slot, CardIV, 0xf0, 1))
        goto exit;

    _arcade_printf("dongle auth 0x01\n");
    if (!card_auth_read(port, slot, CardMaterial, 0xf0, 2))
        goto exit;

......

exit:
    card_auth_60(port, slot);
    mechacon_auth_80(cnum);
    return 0;
}
```

**Comment by israpps** [comment]
arcade mcman increments the port parameter by 2 (mc0 ends up becoming mc2) when calling either SecrAuthCard() or SecrAuthDongle()

**Comment by israpps** [comment]
closing for now. i'll probably re-implement this like I did with #537

**Comment by sp193** [comment]
At some point, Sony seemed to have made the port number +2, perhaps because it refers to the actual SIO2 port which the memory card is connected to. They may have standardized the behaviour to allow MC0 or 1 to be specified, to reduce confusion.

I do recall that this behaviour was present for the functions like sceMcReadPage(), of the early MCMAN versions.

**Pull request closed without merging**